packages/ra-core/src/auth/CanAccess.tsx

```diff
@@ -1,50 +1,53 @@
 import * as React from 'react';
-import { useLoadingContext } from '../core/useLoadingContext';
-import { useUnauthorizedContext } from '../core/useUnauthorizedContext';
 import { useCanAccess, UseCanAccessOptions } from './useCanAccess';
 import { RaRecord } from '../types';
+import { Navigate } from 'react-router';
 
 /**
  * A component that only displays its children after checking whether users are authorized to access the provided resource and action.
  * @param options
  * @param options.action The action to check. One of 'list', 'create', 'edit', 'show', 'delete', or a custom action.
  * @param options.resource The resource to check. e.g. 'posts', 'comments', 'users'
  * @param options.children The component to render if users are authorized.
- * @param options.loading An optional element to render while the authorization is being checked. Defaults to the loading component provided on `Admin`.
- * @param options.unauthorized An optional element to render if users are not authorized. Defaults to the unauthorized component provided on `Admin`.
+ * @param options.loading An optional element to render while the authorization is being checked. Defaults to null.
+ * @param options.unauthorized An optional element to render if users are not authorized. Defaults to null.
```

Comment:
I'd rename it to `accessDenied`. I know it's a pain to do, but since we named the functionality Access Control it makes more sense.